### PR TITLE
example-java native-image runtime crash fix

### PR DESCRIPTION
**Pull request by gradinac**
This PR should fix the runtime crash that occurs when running the `example-java` native-image with a passphrase protected private key.

### CODE REVIEW

**Comment by graemerocher** [comment]
@gradinac Could you sign the CLA?

**Comment by gradinac** [comment]
@graemerocher Done

File: docs-examples/example-java/build.gradle [hunk 1]
```diff
@@ -18,5 +19,5 @@ dependencies {
 mainClassName = "example.Application"
 
 nativeImage {
-    args('--report-unsupported-elements-at-runtime', '--no-fallback', '--no-server')
+    args('--report-unsupported-elements-at-runtime', '--no-fallback', '--no-server', '--rerun-class-initialization-at-runtime=org.bouncycastle.jcajce.provider.drbg.DRBG$Default,org.bouncycastle.jcajce.provider.drbg.DRBG$NonceAndIV')
```
Comment (ilopmar) [inline docs-examples/example-java/build.gradle:22]: Couple of comments:
`--report-unsupported-elements-at-runtime` is discouraged and shouldn't be used unless it's really necessary. Every time I need to use it for a Micronaut app I open an issue on GraalVM

I think `--rerun-class-initialization-at-runtime` was deprecated a long time ago and the flags now are `--initialize-at-run-time` and `--initialize-at-build-time`

Comment (gradinac) [reply docs-examples/example-java/build.gradle:22]: That is correct, but the issue here is that we have to register the `BouncyCastle` provider at build time. This means that any classes that end up along with the provider in the image heap must also be registered at build time. We cannot use the other flags to force the same behavior as `--rerun-class-initialization-at-runtime`.
The alternative here is to use a substitution that will recalculate the values of the specific fields at runtime, but substitutions are a much greater evil than `--rerun-class-initialization-at-runtime` and such be avoided.

Comment (ilopmar) [reply docs-examples/example-java/build.gradle:22]: Ohh, I see. Thanks for the explanation :smiley:

Comment (gradinac) [reply docs-examples/example-java/build.gradle:22]: No problem at all :) The concrete issue here lies in the two classes listed for the class initialization rerun: they have fields that contain a reference to `NativePRNG` on the host system, which we wouldn't have access to on a different system (I assume NativePRNG on Linux is implemented using `/dev/random`).

**Comment by graemerocher** [comment]
I applied these changes manually. Thanks

File: docs-examples/example-java/build.gradle (hunk 1 above)
Comment (cstancu) [reply docs-examples/example-java/build.gradle:22]: Although the `--rerun-class-initialization-at-runtime=` option is deprecated, the concept that it implements is not. You can still do that programmatically through the `org.graalvm.nativeimage.impl.RuntimeClassInitializationSupport#rerunInitialization(java.lang.Class<?>, java.lang.String)` API. We do that in [`SecurityServicesFeature`](https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/SecurityServicesFeature.java#L97) for example.

**Pull request closed without merging**